ulid.go

```diff
@@ -454,3 +461,40 @@ func (id *ULID) Scan(src interface{}) error {
 func (id ULID) Value() (driver.Value, error) {
 	return id.MarshalBinary()
 }
+
+// Monotonic returns an entropy source that is guaranteed to return
+// strictly increasing entropy bytes for the same ULID timestamp.
+// Not safe for concurrent use.
+func Monotonic(entropy io.Reader) io.Reader {
+	return &monotonic{Reader: entropy}
+}
+
+type monotonic struct {
+	io.Reader
+	ms      uint64
+	entropy []byte
+}
+
+func (m *monotonic) MonotonicRead(ms uint64, entropy []byte) (err error) {
+	if len(m.entropy) != 0 && m.ms == ms {
+		err = m.increment()
+		copy(entropy, m.entropy)
+	} else if _, err = io.ReadFull(m, entropy); err == nil {
```

Comment:
io.ReadFull(**m.Reader**, entropy) prevents a bit of ambiguity.

Reply:
👍